### PR TITLE
add color palette to variables and backpack

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Add Color Palette to variables scss file and Backpack.

## WHY
We want to standardize our sitewide color palette.

## HOW
Add all colors from Figma to the variables file, and clean up variables in that file that are no longer being used. Next steps of this project will be pull requests to update everywhere we use the now-named hex codes to refer to the variable.

### Screenshots
Screenshot 2023-08-14 at 1 40 51 PM

### Notion Card Links
https://www.notion.so/quill/Update-color-definitions-on-the-site-to-use-the-ones-from-the-official-color-palette-b58ca8d1d8a64c6cbf5d178b534ee38c?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | N/A - CSS updates
Have you deployed to Staging? | NO - non-app change
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss [hunk 1]
```diff
@@ -123,3 +123,273 @@
...
+    padding: 10px;
+    max-width: 16%;
+    border-bottom: 1px solid #d6d5d6;
+    border-right: 1px solid #d6d5d6;
+    div {
+      width: 40px;
+      height: 40px;
+    }
+  }
+  .quill-black {
```
Comment (happythenewsad) [inline services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss:143]: Have you worked with SCSS maps + `each`? 

This is working scss code, which programmatically generates rules. I used `background-color` as an example:

```css
$quill-colors: quill-green #0AFF0A, quill-red #FF140A;

@each $kvp in $quill-colors {
  .background-color-#{nth($kvp, 1)} {
    background-color: nth($kvp, 2);
  }

}

```

**Review by happythenewsad — COMMENTED** [review]
Proposing a technique that may result in less work/code.

File: services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss (hunk 1 above)
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss:143]: Hey @happythenewsad, I have. I felt like it was easier and more readable to define the variables conventionally in the `variables` file. This file just defines classes for use in the display on the `Backpack` page, not sitewide. I think I rewrite this file as follows:

```
$quill-colors: quill-green $quill-green, quill-red $quill-red

@each $kvp in $quill-colors {
  .#{nth($kvp, 1)} {
    background-color: nth($kvp, 2);
  }
}
```

But that `$quill-colors` variable would just get so long that I kind of prefer the way I have it here. If you feel strongly, though, I can change it.

Comment (happythenewsad) [reply services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss:143]: Ok - you work with this code more than I do, so let's go with your recommendation.

**Review by happythenewsad — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Staff/components/styleGuide/colorPalette.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    title: 'Quill Gold',
+    variableName: "quill-gold"
+  },
+  {
+    title: 'Quill Gold Dark',
+    variableName: "quill-gold-dark"
+  },
+]
+
+const colorNumbers = [null, 50, 20, 10, 5, 1]
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Staff/components/styleGuide/colorPalette.tsx:125]: just curious why we have the first value as `null`?

**Review by eadams17 — APPROVED** [review]
so stoked to have this! thank you for getting this organized

File: services/QuillLMS/client/app/bundles/Staff/components/styleGuide/colorPalette.tsx (hunk 2 above)
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Staff/components/styleGuide/colorPalette.tsx:125]: Just in order to not append anything extra to the title/class name for the first iteration. There may be a more elegant way to do that.

**Pull request merged**